websites/N/NovaStream/presence.ts

```diff
@@ -0,0 +1,93 @@
+const presence = new Presence({
+		clientId: "1334576902081351761",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/sASc3GV.png",
+		smallImageKey: Assets.Play,
+		smallImageText: "Watching free content",
+		startTimestamp: browsingTimestamp,
+		type: ActivityType.Watching,
+	};
+
+	if (document.location.pathname === "/") {
+		const searchInput =
+			document.querySelector<HTMLInputElement>(".search-input")?.value;
+		if (searchInput && searchInput.length > 0)
+			presenceData.state = `Searching for: "${searchInput}"`;
+		else presenceData.state = "Browsing content to watch";
+	} else if (document.location.pathname.includes("/watch")) {
+		let title = document.querySelector("h1.text-2xl")?.textContent,
+			releaseDate,
+			rating,
+			season,
+			episode,
+			runtime,
+			seasonEpisode;
+
+		if (document.location.search.includes("type=movie")) {
+			releaseDate = document.querySelector(
+				"div.space-y-2:nth-child(3) > p:nth-child(2) > span:nth-child(2)"
+			)?.textContent;
+			rating = document
+				.querySelector(
+					"div.space-y-2:nth-child(3) > p:nth-child(4) > span:nth-child(1) > span:nth-child(1)"
+				)
+				?.textContent?.split(" (")[0];
```

Comment:
Try to shorten selectors where possible. For example:
```suggestion
			rating = document
				.querySelector(
					'[x-text*="content?.vote_average?"]'
				)?.textContent
```